### PR TITLE
expose calculation for unit testing

### PR DESCRIPTION
**Pull request by ActionAnalytics**

### CODE REVIEW

File: src/app/day-calculator-modal/day-calculator-modal.component.ts [hunk 1]
```diff
@@ -174,126 +180,152 @@ export class DayCalculatorModalComponent implements OnInit {
...
+      return calcRes;
+  }
+
+  go() {
+    let calcRes = new DayCalculatorResult();
+    try {
+      calcRes = this.calculateStartDate(
+      this.type.value === 'regular',
+      this.type.value === 'suspended',
+      null,
```
Comment (kathydo) [inline src/app/day-calculator-modal/day-calculator-modal.component.ts]: Why null here?

File: src/app/day-calculator-modal/day-calculator-modal.component.ts [hunk 2]
```diff
@@ -174,126 +180,152 @@ export class DayCalculatorModalComponent implements OnInit {
...
-          if (!this.resumeDate) {
-            // Can't find an end date if there is no resume date
-            this.endDate = '';
-            return;
+      } else if (startDate && numDays) {
+        // Find the end date from the start date and number of days
+        endDateMoment = moment(startDate);
+
+        // Include the start date in the calculation.
+        numDays = regular ? 1 : 0;
```
Comment (kathydo) [inline src/app/day-calculator-modal/day-calculator-modal.component.ts]: should this be 
`calcRes.numDays = regular ? 1 : 0;`

File: src/app/day-calculator-modal/day-calculator-modal.component.ts [hunk 3]
```diff
@@ -174,126 +180,152 @@ export class DayCalculatorModalComponent implements OnInit {
...
+
+        // Start counting the days
+        while (numDays < numDays) {
```
Comment (kathydo) [inline src/app/day-calculator-modal/day-calculator-modal.component.ts]: will this work?

File: src/app/day-calculator-modal/day-calculator-modal.component.ts (hunk 2 above)
Comment (ActionAnalytics) [reply src/app/day-calculator-modal/day-calculator-modal.component.ts]: originally was 
numDays = regular ? 1 : 0;

File: src/app/day-calculator-modal/day-calculator-modal.component.ts (hunk 3 above)
Comment (ActionAnalytics) [reply src/app/day-calculator-modal/day-calculator-modal.component.ts]: fixed, thank you, good catch

File: src/app/day-calculator-modal/day-calculator-modal.component.ts [hunk 4]
```diff
@@ -174,126 +180,152 @@ export class DayCalculatorModalComponent implements OnInit {
...
-        // Factor in a suspension
-        if (suspended && this.suspendDate && startDateMoment >= suspendDateMoment) {
-          if (!this.resumeDate) {
-            // Can't find a start date if there is no resume date
-            this.startDate = '';
-            return;
+        // convert moment date back to Date() object so it displays in datepicker
+        calcRes.endDate = new Date (endDateMoment.year(), endDateMoment.month(), endDateMoment.day());
+
+      } else if (endDate && calcRes.numDays) {
```
Comment (kathydo) [inline src/app/day-calculator-modal/day-calculator-modal.component.ts]: we should double check `calcRes.endDate` vs `endDate`

Comment (kathydo) [reply src/app/day-calculator-modal/day-calculator-modal.component.ts]: this should be calcRes.endDate

**Pull request merged**